src/astro/files/base.py

```diff
@@ -0,0 +1,80 @@
+import os
+import pathlib
+from typing import List, Optional
+
+import pandas as pd
+import smart_open
+
+from astro.constants import FileType
+from astro.files.locations import create_file_location
+from astro.files.type import create_type_factory
+
+
+class File:
+    def __init__(self, path: str, conn_id: Optional[str] = None):
+        """Init file object which represent a single file in local/object stores
+
+        :param path: Path to a file in the filesystem/Object stores
+        :param conn_id: Airflow connection ID
+        """
+        self.location = create_file_location(path, conn_id)
+        self.type = create_type_factory(path)
+
+    @property
+    def path(self):
+        return self.location.path
+
+    def get_size(self) -> int:
+        """
+        Return the size (bytes) of the given file.
+
+        :return: File size in bytes
+        :rtype: int
+        """
+        path = pathlib.Path(self.path)
+        return os.path.getsize(path)
+
+    def is_binary(self) -> bool:
+        """
+        Return a FileType given the filepath. Uses a naive strategy, using the file extension.
+
+        :return: True or False
+        :rtype: bool
+        """
+        result: bool = self.type.name == FileType.PARQUET
+        return result
+
+    def write(self, df: pd.DataFrame) -> None:
+        """Write dataframe to all supported files formats
+        :param df: pandas dataframe
```

Comment:
IIR from what @kaxil mentioned, Sphinx expects a blank line between the description and params.

Reply:
@utkarsharma2 I noticed you marked some comments as resolved, but they don't seem updated here - is there any chance the changes were not pushed yet?